core/services/ocrcommon/transmitter.go

```diff
@@ -63,6 +63,47 @@ func NewTransmitter(
 	}, nil
 }
 
+type txManagerOCR2 interface {
+	CreateTransaction(ctx context.Context, txRequest txmgr.TxRequest) (tx txmgr.Tx, err error)
+	GetForwarderForEOAOCR2Feeds(ctx context.Context, eoa, ocr2AggregatorID common.Address) (forwarder common.Address, err error)
+}
+
+type ocr2FeedsTransmitter struct {
+	txManagerOCR2
+	transmitter
+}
+
+// NewOCR2FeedsTransmitter creates a new eth transmitter that handles OCR2 Feeds specific logic surrounding forwarders.
```

Comment:
Would you mind explaining this "specific logic" in a comment here? I'm curious what is special about forwarders in OCR2 Feeds that we need a separate transmitter.

Reply:
Sure, was about to add a description. This change should allow the ocr2 feeds job to always validate its sender. This means checking if forwarders are authorized for EOA and if they are set as a transmitter on the OCR2 aggregator contract. Previously forwarders were chosen once during job start.

Reply:
Although seems like smoke test aren’t passing, hopefully needas a small adjustment